channeldb/payment_control.go

```diff
@@ -217,10 +263,25 @@ func (p *PaymentControl) Success(paymentHash lntypes.Hash,
 			return nil
 		}
 
-		// Record the successful payment info atomically to the
-		// payments record.
-		err = bucket.Put(paymentSettleInfoKey, preimage[:])
-		if err != nil {
+		// We'll also mark all attempts successful, so get the
+		// active attempts bucket.
+		// TODO: this will change.
```

Comment:
Maybe just squash. I personally find it not as easy as one may expect to grasp these intermediate refactor states. I'd rather look at the whole diff even if it is longer (and the total diff is probably slightly smaller)